File: shamir.cpp
#include "decrypt.h"
#include "encrypt.h"
#include "ssssen.h"
#include "shamir.h"
#include <cstdlib>
#include <filesystem>
#include <iostream>


namespace fs = std::filesystem;



void show_usage(int argc, char** argv) {
    std::cerr << "Usage:\n"
        << "  " << argv[0] << " e(ncrypt) <file_path>\n"
        << "  " << argv[0] << " d(ecrypt) <file_path>\n";
}


int shamir() {
    std::string file_path;
    char mode;

    std::cout << "Enter mode (e - encrypt, d - decrypt): ";
    std::cin >> mode;

    std::cout << "Enter file path: ";
    std::cin >> file_path;

    if (mode == 'e') {
        int res = encrypt(file_path);
        if (res != 0) {
            std::cerr << "Error encrypting: " << file_path << std::endl;
        }
    } else if (mode == 'd') {
        decrypt(file_path);
    } else {
        std::cerr << "Unknown mode. Use 'e' or 'd'.\n";
    }

    return 0;
}

File: Shamir.java
import java.util.Scanner;

public class Shamir {
    public static void run() {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter mode (e - encrypt, d - decrypt): ");
        char mode = sc.next().charAt(0);
        if (mode == 'e') {
            Encrypt.run();
        } else if (mode == 'd') {
            Decrypt.run();
        } else {
            System.out.println("Unknown mode");
        }
    }
}
